===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ p11_index_take:n$17 == { 0 } }) free(*(p11_index_take:obj)) at 11 (line 520, column 7) }
===== Actual patch in diff format =====
@@ -518,6 +518,7 @@ p11_index_take (p11_index *index,
 	return_val_if_fail (obj->attrs != NULL, CKR_GENERAL_ERROR);
 
 	if (!p11_dict_set (index->objects, &obj->handle, obj))
+		free(obj);
 		return_val_if_reached (CKR_HOST_MEMORY);
 
 	index_hash (index, obj);
